Unable to create service

### Checklist

- [x] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [x] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [x] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [x] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [x] This issue contains only one bug.
- [x] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.28.0-25-202508180153

### Steps to reproduce the bug

1. Click on the link : [notify](https://youtu.be/OZRG7Og61mw?utm_campaign=2025-08-19-youtube-first-launch-teaser%20%2801K318CDH30KCJ5KNDG94SBWPE%29&utm_medium=email&utm_source=Newsletter&_kx=a1m7o-dnv-0MpzU_3UTqUwQmwFH84-rmErkoTiMc53DPD2-WmiAd9TicS7DiGipJ.LNYsuiLNYsui) 
2. That's all

### Expected behavior

Go to a Youtube video

### Actual behavior

Crash : Unable to create service 

### Screenshots/Screen recordings

_No response_

### Logs

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ FR
* __Content Language:__ fr-FR
* __App Language:__ fr_FR
* __Service:__ none
* __Timestamp:__ 2025-08-19T22:46:52.098+02:00
* __Package:__ org.schabi.newpipe.refactor.nightly
* __Service:__ none
* __Version:__ 0.28.0-25-202508180153
* __OS:__ Linux Android 15 - 35
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to create service org.schabi.newpipe.RouterActivity$FetcherService: android.app.MissingForegroundServiceTypeException: Starting FGS without a type  callerApp=ProcessRecord{9749d2 20930:org.schabi.newpipe.refactor.nightly/u0a202} targetSDK=35
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:5148)
	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2547)
	at android.os.Handler.dispatchMessage(Handler.java:109)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8934)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:911)
Caused by: android.app.MissingForegroundServiceTypeException: Starting FGS without a type  callerApp=ProcessRecord{9749d2 20930:org.schabi.newpipe.refactor.nightly/u0a202} targetSDK=35
	at android.app.MissingForegroundServiceTypeException$1.createFromParcel(MissingForegroundServiceTypeException.java:53)
	at android.app.MissingForegroundServiceTypeException$1.createFromParcel(MissingForegroundServiceTypeException.java:49)
	at android.os.Parcel.readParcelableInternal(Parcel.java:5094)
	at android.os.Parcel.readParcelable(Parcel.java:5076)
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3256)
	at android.os.Parcel.createException(Parcel.java:3245)
	at android.os.Parcel.readException(Parcel.java:3228)
	at android.os.Parcel.readException(Parcel.java:3170)
	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7268)
	at android.app.Service.startForeground(Service.java:776)
	at org.schabi.newpipe.RouterActivity$FetcherService.onCreate(RouterActivity.java:949)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:5135)
	... 9 more

```
</details>
<hr>


### Affected Android/Custom ROM version

_No response_

### Affected device model

_No response_

### Additional information

_No response_

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -98,8 +98,14 @@
 
         <service android:name=".local.subscription.services.SubscriptionsImportService" />
         <service android:name=".local.subscription.services.SubscriptionsExportService" />
-        <service android:name=".local.feed.service.FeedLoadService"
+        <service
+            android:name=".local.feed.service.FeedLoadService"
             android:foregroundServiceType="dataSync" />
+
+        <service
+            android:name="androidx.work.impl.foreground.SystemForegroundService"
+            android:foregroundServiceType="dataSync"
+            tools:node="merge" />
 
         <activity
             android:name=".PanicResponderActivity"

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -96,8 +96,14 @@
             android:exported="false"
             android:label="@string/title_activity_about" />
 
-        <service android:name=".local.subscription.services.SubscriptionsImportService" />
-        <service android:name=".local.subscription.services.SubscriptionsExportService" />
+        <service
+            android:name=".local.subscription.services.SubscriptionsImportService"
+            android:foregroundServiceType="dataSync" />
+
+        <service
+            android:name=".local.subscription.services.SubscriptionsExportService"
+            android:foregroundServiceType="dataSync" />
+
         <service
             android:name=".local.feed.service.FeedLoadService"
             android:foregroundServiceType="dataSync" />
@@ -431,6 +437,7 @@
         </activity>
         <service
             android:name=".RouterActivity$FetcherService"
+            android:foregroundServiceType="dataSync"
             android:exported="false" />
 
         <!-- opting out of sending metrics to Google in Android System WebView -->

--- a/app/src/main/java/org/schabi/newpipe/local/feed/notifications/NotificationWorker.kt
+++ b/app/src/main/java/org/schabi/newpipe/local/feed/notifications/NotificationWorker.kt
@@ -2,7 +2,6 @@ package org.schabi.newpipe.local.feed.notifications
 
 import android.content.Context
 import android.content.pm.ServiceInfo
-import android.os.Build
 import android.util.Log
 import androidx.core.app.NotificationCompat
 import androidx.work.Constraints
@@ -85,7 +84,8 @@ class NotificationWorker(
             .setPriority(NotificationCompat.PRIORITY_LOW)
             .setContentTitle(applicationContext.getString(R.string.feed_notification_loading))
             .build()
-        val serviceType = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) ServiceInfo.FOREGROUND_SERVICE_TYPE_DATA_SYNC else 0
+        @Suppress("NewApi") // ServiceInfo constant is inlined
+        val serviceType = ServiceInfo.FOREGROUND_SERVICE_TYPE_DATA_SYNC
         setForegroundAsync(ForegroundInfo(FeedLoadService.NOTIFICATION_ID, notification, serviceType))
     }
 

--- a/app/src/main/java/org/schabi/newpipe/local/feed/notifications/NotificationWorker.kt
+++ b/app/src/main/java/org/schabi/newpipe/local/feed/notifications/NotificationWorker.kt
@@ -2,6 +2,7 @@ package org.schabi.newpipe.local.feed.notifications
 
 import android.content.Context
 import android.content.pm.ServiceInfo
+import android.os.Build
 import android.util.Log
 import androidx.core.app.NotificationCompat
 import androidx.work.Constraints
@@ -84,8 +85,7 @@ class NotificationWorker(
             .setPriority(NotificationCompat.PRIORITY_LOW)
             .setContentTitle(applicationContext.getString(R.string.feed_notification_loading))
             .build()
-        @Suppress("NewApi") // ServiceInfo constant is inlined
-        val serviceType = ServiceInfo.FOREGROUND_SERVICE_TYPE_DATA_SYNC
+        val serviceType = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) ServiceInfo.FOREGROUND_SERVICE_TYPE_DATA_SYNC else 0
         setForegroundAsync(ForegroundInfo(FeedLoadService.NOTIFICATION_ID, notification, serviceType))
     }
 

--- a/app/src/main/java/org/schabi/newpipe/local/feed/notifications/NotificationWorker.kt
+++ b/app/src/main/java/org/schabi/newpipe/local/feed/notifications/NotificationWorker.kt
@@ -85,6 +85,7 @@ class NotificationWorker(
             .setPriority(NotificationCompat.PRIORITY_LOW)
             .setContentTitle(applicationContext.getString(R.string.feed_notification_loading))
             .build()
+        // ServiceInfo constants are not used below Android Q, so 0 is set here
         val serviceType = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) ServiceInfo.FOREGROUND_SERVICE_TYPE_DATA_SYNC else 0
         setForegroundAsync(ForegroundInfo(FeedLoadService.NOTIFICATION_ID, notification, serviceType))
     }

--- a/app/src/main/java/org/schabi/newpipe/local/feed/notifications/NotificationWorker.kt
+++ b/app/src/main/java/org/schabi/newpipe/local/feed/notifications/NotificationWorker.kt
@@ -1,6 +1,8 @@
 package org.schabi.newpipe.local.feed.notifications
 
 import android.content.Context
+import android.content.pm.ServiceInfo
+import android.os.Build
 import android.util.Log
 import androidx.core.app.NotificationCompat
 import androidx.work.Constraints
@@ -83,7 +85,8 @@ class NotificationWorker(
             .setPriority(NotificationCompat.PRIORITY_LOW)
             .setContentTitle(applicationContext.getString(R.string.feed_notification_loading))
             .build()
-        setForegroundAsync(ForegroundInfo(FeedLoadService.NOTIFICATION_ID, notification))
+        val serviceType = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) ServiceInfo.FOREGROUND_SERVICE_TYPE_DATA_SYNC else 0
+        setForegroundAsync(ForegroundInfo(FeedLoadService.NOTIFICATION_ID, notification, serviceType))
     }
 
     companion object {

--- a/app/src/main/java/org/schabi/newpipe/player/notification/NotificationUtil.java
+++ b/app/src/main/java/org/schabi/newpipe/player/notification/NotificationUtil.java
@@ -174,12 +174,8 @@ public final class NotificationUtil {
         }
         updateNotification();
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            player.getService().startForeground(NOTIFICATION_ID, notificationBuilder.build(),
-                    ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PLAYBACK);
-        } else {
-            player.getService().startForeground(NOTIFICATION_ID, notificationBuilder.build());
-        }
+        ServiceCompat.startForeground(player.getService(), NOTIFICATION_ID,
+                notificationBuilder.build(), ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PLAYBACK);
     }
 
     public void cancelNotificationAndStopForeground() {

--- a/app/src/main/java/org/schabi/newpipe/player/notification/NotificationUtil.java
+++ b/app/src/main/java/org/schabi/newpipe/player/notification/NotificationUtil.java
@@ -167,15 +167,16 @@ public final class NotificationUtil {
                 && notificationBuilder.mActions.get(2).actionIntent != null);
     }
 
-    @SuppressLint("NewApi") // ServiceInfo constant is inlined
     public void createNotificationAndStartForeground() {
         if (notificationBuilder == null) {
             notificationBuilder = createNotification();
         }
         updateNotification();
 
+        final int serviceType = Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q
+                ? ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PLAYBACK : 0;
         ServiceCompat.startForeground(player.getService(), NOTIFICATION_ID,
-                notificationBuilder.build(), ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PLAYBACK);
+                notificationBuilder.build(), serviceType);
     }
 
     public void cancelNotificationAndStopForeground() {

--- a/app/src/main/java/org/schabi/newpipe/player/notification/NotificationUtil.java
+++ b/app/src/main/java/org/schabi/newpipe/player/notification/NotificationUtil.java
@@ -167,7 +167,7 @@ public final class NotificationUtil {
                 && notificationBuilder.mActions.get(2).actionIntent != null);
     }
 
-
+    @SuppressLint("NewApi") // ServiceInfo constant is inlined
     public void createNotificationAndStartForeground() {
         if (notificationBuilder == null) {
             notificationBuilder = createNotification();

--- a/app/src/main/java/org/schabi/newpipe/player/notification/NotificationUtil.java
+++ b/app/src/main/java/org/schabi/newpipe/player/notification/NotificationUtil.java
@@ -173,6 +173,7 @@ public final class NotificationUtil {
         }
         updateNotification();
 
+        // ServiceInfo constants are not used below Android Q, so 0 is set here
         final int serviceType = Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q
                 ? ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PLAYBACK : 0;
         ServiceCompat.startForeground(player.getService(), NOTIFICATION_ID,

